### PR TITLE
Add an `allowed_classes` parameter to ammonia

### LINKED ISSUE #24: A way to specify approved *values* of attributes
Use case: I would like to allow syntax highlighting classes on `code` tags but not [arbitrary classes](https://github.com/ChALkeR/notes/blob/master/Improper-markup-sanitization.md). 

I'd be fine having this ability just for `class` attribute values, as [sanitize-html does](https://www.npmjs.com/package/sanitize-html#allowed-css-classes); not sure if there's a use case for values of any attribute or not.

### PR DESCRIPTION
**Pull request by Eijebong**
Fixes #24

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -238,18 +245,33 @@ impl<'a> Ammonia<'a> {
...
-            if let &Some(ref base) = url_base {
+            if self.allowed_classes.contains_key(&*name.local) {
                 for attr in &mut *attrs.borrow_mut() {
-                    if is_url_attr(&*name.local, &*attr.name.local) {
-                        let url = base.join(&*attr.value).expect("invalid URLs should be stripped earlier");
-                        attr.value = format_tendril!("{}", url);
+                    if &attr.name.local == "class" {
+                        let mut classes = vec![];
+                        for class in attr.value.split(' ') {
+                            if self.allowed_classes[&*name.local].contains(class) {
```
Comment (lnicola) [inline src/lib.rs]: Can you cache this? And does it work if the element is not present in the map?

**Comment by lnicola** [comment]
Sorry if I missed something, I'm in my phone. I don't understand how that line works when the hashmap is empty. I expected the other tests to fail.

Looks good otherwise, please ignore this if I'm wrong.

File: src/lib.rs (hunk 1 above)
Comment (Eijebong) [inline src/lib.rs]: Well we check that the element is in the map above. `if self.allowed_classes.contains_key(&*name.local)`

Maybe I can do something better with .get() and pattern matching, I'll try that :)

Comment (lnicola) [inline src/lib.rs]: Oh, I see. Then yeah, maybe it would be better to only look that up once.

Comment (Eijebong) [inline src/lib.rs]: Yep, done, thanks

**Comment by notriddle** [comment]
I fixed a bug related to rel and URL base rewriting.

**Comment by notriddle** [comment]
See https://github.com/notriddle/ammonia/pull/25/commits/17e14da4a419954f852571b853730120984ffc1a?w=1 for a decent description of what I changed (the `?w=1` is important)

**Comment by Eijebong** [comment]
Woups, nice catch

**Comment by notriddle** [comment]
Does this look good enough to merge now?

**Comment by lnicola** [comment]
Looks good to me.

**Comment by lnicola** [comment]
> They said in a comment that it was good, but forgot to end their review.

Sorry for that, phone and I didn't realize the review was still open :(.

**Pull request merged**